src/ucp/core/ucp_am.c

```diff
@@ -1177,12 +1177,17 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_recv_data_nbx,
         ret = UCS_STATUS_PTR(status);
     }
 
-    /* Clear this flag, because receive operation is already completed and desc
-     * is not needed anymore. If receive operation was invoked from UCP AM
-     * callback, UCT AM handler would release this desc (by returning UCS_OK)
-     * back to UCT.
-     */
-    desc->flags &= ~UCP_RECV_DESC_FLAG_AM_CB_INPROGRESS;
+    if ((status != UCS_INPROGRESS) &&
```

Comment:
no need to check for `INPROGRESS`, can add assert that `status != INPROGESS` instead